docs/model_developers/release.rst

```diff
@@ -36,42 +36,59 @@ When done, first use git to add files:
 
     git add *
 
-To commit, use the module tool. This will create a new dev release.
+Create a new dev release:
 
 .. code-block:: bash
 
-    inmanta module commit --patch -m "Fixed small bug"
+    inmanta module release --dev --patch -m "Fixed small bug"
 
-This command will set the version to the next dev version (`+0.0.1dev`) and add a timestamp.
+This command will set the version to the next dev version, e.g. ``+0.0.1dev`` here for a patch increment.
 
 The module tool supports semantic versioning.
-Use one of ``--major``, ``--minor`` or ``--patch`` to update version numbers: <major>.<minor>.<patch>
+Use one of ``--major``, ``--minor`` or ``--patch`` to update version numbers: ``<major>.<minor>.<patch>``
 
 For the dev releases, no tags are created.
+Once the dev version is ready for release, perform a proper release by following
+the steps in the `Release Versions`_ section below.
 
 Release Versions
 ----------------
 
-To make an actual release (without `.dev` at the end):
+To perform an actual non-dev release:
+
+First, merge your changes from your feature branch (e.g. ``mywork`` in the previous example) onto the branch dedicated to
+releases (e.g. master) and then do the actual release:
 
 .. code-block:: bash
 
-    inmanta module commit -r -m "First Release"
+    git checkout master
+    git pull
+    inmanta module release
+    git push
+    git push origin {tag}
+
+This will remove the ``.dev`` version and automatically set the right tags on the module.
+
+To automatically freeze all dependencies of this module to the currently checked out versions:
+
+For v1 modules:
+
+.. code-block:: bash
 
-This will remove the `.dev` version and automatically set the right tags on the module.
+    inmanta module freeze --recursive --operator "=="
 
-To automatically freeze all dependencies of this module to the currently checked out versions
+For v2 modules:
 
 .. code-block:: bash
 
-	inmanta module freeze --recursive --operator ==
+    pip freeze > out_file.txt
```

Comment:
Do I understand correctly that `inmanta module freeze` does not work for v2?

Either way, I don't think this is a helpful substitute because it will not be used by the orchestrator. I'm not entirely certain what the motivation is for freezing a module's dependencies. If it's just for repititive testing this is not an issue, but I would guess it's to freeze it on the orchestrator. Then again, with the project freeze available, I don't see the value. Perhaps this section is just no longer relevant?

@wouterdb could you shed some light on this?

Reply:
> Do I understand correctly that inmanta module freeze does not work for v2?

Yes I think it's not intended to work for v2 :
https://github.com/inmanta/inmanta-core/blob/5c955d1278fe8d90440e598aa075526225adcce7/src/inmanta/moduletool.py#L664
and
https://github.com/inmanta/inmanta-core/blob/5c955d1278fe8d90440e598aa075526225adcce7/src/inmanta/moduletool.py#L1104

> Perhaps this section is just no longer relevant?

I think so, I'll update the relevant parts of `inmanta module freeze` for v2 modules depending on @wouterdb 's answer

Reply:
As discussed in standup:
- phase out `inmanta module freeze` in iso8: https://github.com/inmanta/inmanta-core/issues/7076